src/System.Private.ServiceModel/tools/scripts/WcfTestServerSetup.cmd

```diff
@@ -0,0 +1,191 @@
+    echo Could not find: %appcmd%.
+    echo Please make sure IIS is installed.
+    goto :Failure
+)
+
+:: Clean up any existing setup
+echo Deleting WCF repo at %CurrentRepo% if exists and associated application pools and services hosted on IIS...
+if exist %CurrentRepo% rmdir /s /q %CurrentRepo%
+if exist %WcfTestDir% if /I '%MasterRepo%'=='%CurrentRepo%' rmdir /s /q %WcfTestDir%
+%appcmd% delete app "Default Web Site/%PRServiceName%" >nul
+%appcmd% delete apppool %PRServiceName% >nul
+%appcmd% delete app "Default Web Site/%WcfServiceName%" >nul
+%appcmd% delete apppool %WcfServiceName% >nul
+echo Clean up done.
+if /I '%2'=='/c' goto :Done
+
+:: Make sure git.exe is available
+where git.exe /Q
+if ERRORLEVEL 1 (
+    echo Could not find: git.exe.
+    echo Please make sure git is installed and its location is part of PATH environment.
+    goto :Failure
+)
+
+:: Clone a WCF repo
+set cmd=git clone https://github.com/dotnet/wcf %CurrentRepo%
+call %cmd%
+if ERRORLEVEL 1 goto :Failure
+pushd %CurrentRepo%
+call :Run git config --add origin.fetch "+refs/pull/*/head:refs/remotes/origin/pr/*"
+if ERRORLEVEL 1 goto :Failure
+popd
+
+:: Create a new application pool and an application for the WCF test service
+echo Create IIS application pool: %WcfServiceName%
+call :Run %appcmd% add apppool /name:%WcfServiceName% /processModel.IdentityType:LocalSystem /managedRuntimeVersion:v4.0 /managedPipelineMode:Integrated
+if ERRORLEVEL 1 goto :Failure
+echo Create IIS application: %WcfServiceName%
+call :Run %appcmd% add app /site.name:"Default Web Site" /path:/%WcfServiceName% /physicalPath:%CurrentRepo%\src\System.Private.ServiceModel\tools\IISHostedWcfService /applicationPool:%WcfServiceName% /enabledProtocols:"http,net.tcp"
+if ERRORLEVEL 1 goto :Failure
+
+:: Grant app pool %WcfServiceName% "Read and Execute" access to the IISHostedWcfService and its subdirectories
+echo Grant app pool %WcfServiceName% "Read and Execute" access to the IISHostedWcfService and its subdirectories
+call :Run icacls %CurrentRepo%\src\System.Private.ServiceModel\tools\IISHostedWcfService /grant:r "IIS APPPOOL\%WcfServiceName%":RX /T /Q
+if ERRORLEVEL 1 goto :Failure
+
+:: Setup PR service only if this is a master repo
+:: Uncomment line below when we switch to use one central PRService
+REM if /I NOT '%MasterRepo%'=='%CurrentRepo%' goto :SkipPRService
+
+:: Create a new application pool and an application for the PR service
+echo Create IIS applicaton pool: %PRServiceName%
+call :Run %appcmd% add apppool /name:%PRServiceName% /managedRuntimeVersion:v4.0 /managedPipelineMode:Integrated
+if ERRORLEVEL 1 goto :Failure
+echo Create IIS applicatoin: %PRServiceName%
+call :Run %appcmd% add app /site.name:"Default Web Site" /path:/%PRServiceName% /physicalPath:%CurrentRepo%\src\System.Private.ServiceModel\tools\PRService -applicationPool:%PRServiceName%
+if ERRORLEVEL 1 goto :Failure
+
+:: Grant app pool %PRServiceName% "Modify" access to %CurrentRepo% and its subdirectories
```

Comment:
Why not full access?  If git syncs, we should anticipate its automated GC to delete files at unexpected times.  When we sync or clean, we should expect files to be both deleted and created.